Bottom: 3ac02ca0ee786e913d18ea9852678a2b6a8d5279
Top:    c8967588920b9a9b24fef318c586aa32d62fd823
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 13:09:33 +0100

Refresh of hid-gtk-use-gtk_widget_get_win

---

diff --git a/configure.ac b/configure.ac
index 71ab1c4..141b5d2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -770,6 +770,8 @@ $GTKGLEXT_PKG_ERRORS])]
 	GTKGLEXT_VER=`$PKG_CONFIG gtkglext-1.0 --modversion`
 	fi
 
+	GTK_CFLAGS="$GTK_CFLAGS -DGSEAL_ENABLE"
+
 	;;
 
       gcode|nelma|png )
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 5fa2d9c..5384440 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -261,7 +261,8 @@ ghid_shift_is_pressed ()
   if( ! ghid_gui_is_up ) 
     return 0;
 
-  gdk_window_get_pointer (out->drawing_area->window, NULL, NULL, &mask);
+  gdk_window_get_pointer (gtk_widget_get_window (out->drawing_area),
+                          NULL, NULL, &mask);
   return (mask & GDK_SHIFT_MASK) ? TRUE : FALSE;
 }
 
@@ -274,7 +275,8 @@ ghid_control_is_pressed ()
   if( ! ghid_gui_is_up )
     return 0;
 
-  gdk_window_get_pointer (out->drawing_area->window, NULL, NULL, &mask);
+  gdk_window_get_pointer (gtk_widget_get_window (out->drawing_area),
+                          NULL, NULL, &mask);
   return (mask & GDK_CONTROL_MASK) ? TRUE : FALSE;
 }
 
@@ -287,7 +289,8 @@ ghid_mod1_is_pressed ()
   if( ! ghid_gui_is_up )
     return 0;
 
-  gdk_window_get_pointer (out->drawing_area->window, NULL, NULL, &mask);
+  gdk_window_get_pointer (gtk_widget_get_window (out->drawing_area),
+                          NULL, NULL, &mask);
 #ifdef __APPLE__
   return (mask & ( 1 << 13 ) ) ? TRUE : FALSE;  // The option key is not MOD1, although it should be...
 #else
@@ -326,7 +329,8 @@ ghid_set_crosshair (int x, int y, int action)
    * and gdk_display_warp_pointer work relative to the whole display, whilst
    * our coordinates are relative to the drawing area origin.
    */
-  gdk_window_get_origin (gport->drawing_area->window, &offset_x, &offset_y);
+  gdk_window_get_origin (gtk_widget_get_window (gport->drawing_area),
+                         &offset_x, &offset_y);
   display = gdk_display_get_default ();
 
   switch (action) {
@@ -1544,7 +1548,8 @@ Benchmark (int argc, char **argv, Coord x, Coord y)
   do
     {
       ghid_invalidate_all ();
-      gdk_window_process_updates (gport->drawing_area->window, FALSE);
+      gdk_window_process_updates (gtk_widget_get_window (gport->drawing_area),
+                                  FALSE);
       time (&end);
       i++;
     }
@@ -1595,7 +1600,8 @@ Center(int argc, char **argv, Coord pcb_x, Coord pcb_y)
    */
 
   ghid_pcb_to_event_coords (pcb_x, pcb_y, &widget_x, &widget_y);
-  gdk_window_get_origin (gport->drawing_area->window, &offset_x, &offset_y);
+  gdk_window_get_origin (gtk_widget_get_window (gport->drawing_area),
+                         &offset_x, &offset_y);
 
   pointer_x = offset_x + widget_x;
   pointer_y = offset_y + widget_y;
